api/client/registry.go

```diff
@@ -148,6 +149,34 @@ func (cli *DockerCli) ConfigureAuth(flUser, flPassword, serverAddress string, is
 	return authconfig, nil
 }
 
+// ResolveAuthConfigFromImage retrieves that AuthConfig using the image string
+func (cli *DockerCli) ResolveAuthConfigFromImage(ctx context.Context, image string) (types.AuthConfig, error) {
```

Comment:
If this func is not used elsewhere you can combine it with `RetrieveAuthTokenFromImage` or make it private.